app/models/host.rb

```diff
@@ -11,15 +11,14 @@ def self.method_missing(method, *args, &block)
         type = args[0][:type]   # stores the type for later usage.
       end
     end
-
     type.constantize.send(method,*args, &block)
   end
 
   # the API base controller expects to call 'respond_to?' on this, which
   # this module doesn't have. So we stub it out to make that logic work for
   # the "find_by_*" classes that Rails will provide
   def self.respond_to?(method, include_private = false)
-    if method.to_s =~ /\Afind_by_(.*)\Z/
+    if method.to_s =~ /\Afind_by_(.*)\Z/ || method == :reorder
```

Comment:
[Scope search ](https://github.com/wvanbergen/scoped_search/blob/master/lib/scoped_search/auto_complete_builder.rb#L222) checks if the field class responds to `:reorder`, Host class returns a false answer, though it can run it thanks to the `method_missing` implement